smartpy/optimizers/optimizer.py

```diff
@@ -0,0 +1,26 @@
+import theano.tensor as T
+
+
+class Optimizer(object):
+    def __init__(self, model, loss_fct, dataset, update_rules=None, param_modifiers=None):
+        self.model = model
+        self.loss = loss_fct
+        self.dataset = dataset
+
+        self.update_rules = update_rules if update_rules is not None else []
+        self.param_modifiers = param_modifiers if param_modifiers is not None else []
+
+        self.data = [dataset.inputs_shared, dataset.targets_shared]
+        self.inputs = [T.matrix('input' + str(i)) for i in range(len(self.data))]
+        self.loss = loss_fct(*self.inputs)
+
+    def append_update_rule(self, update_rule):
+        self.update_rules.append(update_rule)
```

Comment:
Should be private  and there should probably be some type checking ... let's look at python standards and good practices for that.